XenAdmin/Wizards/PatchingWizard/PatchingWizard_UploadPage.cs

```diff
@@ -509,6 +509,7 @@ private void singleAction_Completed(ActionBase sender)
                                 AllIntroducedPoolUpdates.Add(PoolUpdate, SelectedNewPatchPath);
                             }
                         }
+                        srUploadedUpdates = (action as UploadSupplementalPackAction).srUploadedUpdates;

```

Comment:
By assigning `srUploadedUpdates` like this, it will always contain only the SRs from the last upload action. If I am uploading the update to multiple pools, we will have multiple upload actions here, but with this code only the SRs from one action are saved (from the one that finished last, I assume).
So we will need to add to the list instead of replacing it.

Reply:
I think we need `srUploadedUpdates` to be a `Dictionary<Host, SR>`  instead of a list containing all the SRs where we uploaded the patch, because the prechecks are run per host and we shouldn't check if all these SRs are broken for each host. We should only check the SRs that are relevant to that host,